scripts/globals/wsquest.lua

```diff
@@ -0,0 +1,562 @@
+            ['finish'] = 450
+        },
+        ['options'] =
+        {
+            ['acceptStart'] = nil,
+            ['dropped'] = 1,
+            ['abandon'] = 3
+        }
+    },
+    ----------------------------------------
+    -- Quest: From Saplings Grow
+    --   NPC: Perih Vashai (Windurst Woods)
+    --    WS: Empyrean Arrow (Archery)
+    ----------------------------------------
+    ['fromSaplingsGrow'] =
+    {
+        ['wsId'] = 199,
+        ['wsUnlockId'] = 13,
+        ['skillId'] = dsp.skill.ARCHERY,
+        ['minSkill'] = 250,
+        ['trialWeaponId'] = 18144,
+        ['logId'] = WINDURST,
+        ['questId'] = FROM_SAPLINGS_GROW,
+        ['fameRegion'] = WINDURST,
+        ['eventIds'] =
+        {
+            ['start'] = 661,
+            ['cont1'] = 662,
+            ['cont2'] = 665,
+            ['tradedUnfinishedWeapon'] = 663,
+            ['tradedFinishedWeapon'] = 664,
+            ['finish'] = 666
+        },
+        ['options'] =
+        {
+            ['acceptStart'] = nil,
+            ['dropped'] = 1,
+            ['abandon'] = 3
+        }
+    },
+    -------------------------------------------
+    -- Quest: Shoot First, Ask Questions Later
+    --   NPC: Cid (Metalworks)
+    --    WS: Detonator (Marksmanship)
+    -------------------------------------------
+    ['shootFirstAskQuestionsLater'] =
+    {
+        ['wsId'] = 215,
+        ['wsUnlockId'] = 14,
+        ['skillId'] = dsp.skill.MARKSMANSHIP,
+        ['minSkill'] = 250,
+        ['trialWeaponId'] = 18146,
+        ['logId'] = BASTOK,
+        ['questId'] = SHOOT_FIRST_ASK_QUESTIONS_LATER,
+        ['fameRegion'] = BASTOK,
+        ['eventIds'] =
+        {
+            ['start'] = 795,
+            ['cont1'] = 796,
+            ['cont2'] = nil,
+            ['tradedUnfinishedWeapon'] = 797,
+            ['tradedFinishedWeapon'] = 798,
+            ['finish'] = 799
+        },
+        ['options'] =
+        {
+            ['acceptStart'] = nil,
+            ['dropped'] = 1,
+            ['abandon'] = 2
+        }
+    },
+}
+
+local WSQUEST_START = 0 -- Quest is available or completed
+local WSQUEST_CONT1 = 1 -- Player has accepted quest ('cont1')
+local WSQUEST_CONT2 = 2 -- Player has turned in completed trial weapon and received Map to Annals ('cont2')
+local WSQUEST_FINISH = 3 -- Player has killed NM and received Annals of Truth ('finish')
+
+local function getQuestState(quest,player)
+    if (player:getQuestStatus(quest.logId,quest.questId) == QUEST_ACCEPTED) then
+        if (player:hasKeyItem(dsp.ki.ANNALS_OF_TRUTH)) then
+            return WSQUEST_FINISH
+        elseif (player:hasKeyItem(dsp.ki.MAP_TO_THE_ANNALS_OF_TRUTH)) then
+            return WSQUEST_CONT2
+        else
+            return WSQUEST_CONT1
+        end
+    else
+        local canEquip = player:canEquipItem(quest.trialWeaponId,true)
+        local sufficientSkill = player:getSkillLevel(quest.skillId) >= quest.minSkill
```

Comment:
Doesn't getSkillLevel return the current skill value, ie. including merits and other modifiers? Actually, does retail check for base skill level or current for WS quests? If it's the former, IIRC there's a function that returns the base skill level, but I can't remember what it was.

Reply:
You're right, getSkillLevel, I forgot it returns the total skill the character has (merits, gear, etc).

I just tested this on retail with Great Axe, native skill is 228 but with merits I have 240 effective skill, and Iron Eater will not give me the quest, so we'll have to change this.

Reply:
Took a look around, I believe the function you'll need is getCharSkillLevel()